src/dyn/libdyn.cpp

```diff
@@ -375,7 +375,7 @@ void export_dyn_decoherence_objects(){
   int instantaneous_decoherence_variant, int collapse_option) = &instantaneous_decoherence;
   def("instantaneous_decoherence", expt_instantaneous_decoherence_v1);
 
-  void (*expt_instantaneous_decoherence_dia_v1)(CMATRIX& Coeff, nHamiltonian& ham,
+  void (*expt_instantaneous_decoherence_dia_v1)(CMATRIX& Coeff,

```

Comment:
even though the ham was not used in this function, it is there for a reason - since the function is only experimental, there maybe need to pass some Hamiltonian-related information to this function, so I did it only as a placeholder; also, if you change this signature here, one would need to also adjust the corresponding interface and implementation in the .h and .cpp files.